web-registry/src/pages/Home/Home.loader.ts

```diff
@@ -0,0 +1,74 @@
+import { QueryClient } from '@tanstack/react-query';
+
+import { getEcocreditQueryClient } from 'lib/clients/ecocreditQueryClient';
+import { getMarketplaceQueryClient } from 'lib/clients/marketplaceQueryClient';
+import { getSimplePriceQuery } from 'lib/queries/react-query/coingecko/simplePrice/simplePriceQuery';
+import { getProjectsQuery } from 'lib/queries/react-query/ecocredit/getProjectsQuery/getProjectsQuery';
+import { getAllowedDenomQuery } from 'lib/queries/react-query/marketplace/getAllowedDenomQuery/getAllowedDenomQuery';
+import { getSellOrdersExtentedQuery } from 'lib/queries/react-query/marketplace/getSellOrdersExtentedQuery/getSellOrdersExtentedQuery';
+import { getAllCreditClassesQuery } from 'lib/queries/react-query/sanity/getAllCreditClassesQuery/getAllCreditClassesQuery';
+import { getAllHomePageQuery } from 'lib/queries/react-query/sanity/getAllHomePageQuery/getAllHomePageQuery';
+import { getFromCacheOrFetch } from 'lib/queries/react-query/utils/getFromCacheOrFetch';
+
+import { client as sanityClient } from '../../sanity';
+
+type LoaderType = {
+  queryClient: QueryClient;
+};
+
+export const homeLoader =
+  ({ queryClient }: LoaderType) =>
+  async ({ params }: { params: any }) => {
+    // Clients
+    const ecocreditClient = await getEcocreditQueryClient();
+    const marketplaceClient = await getMarketplaceQueryClient();
+
+    // Queries
+    const allHomePageQuery = getAllHomePageQuery({ sanityClient });
+    const allCreditClassesQuery = getAllCreditClassesQuery({ sanityClient });
+    const simplePriceQuery = getSimplePriceQuery({});
+    const projectsQuery = getProjectsQuery({
+      client: ecocreditClient,
+      request: {},
+    });
+    const sellOrdersQuery = getSellOrdersExtentedQuery({
+      client: marketplaceClient,
+      reactQueryClient: queryClient,
+      request: {},
+    });
+    const allowedDenomQuery = getAllowedDenomQuery({
+      client: marketplaceClient,
+    });
+
+    // Fetch or Cache
+
+    // Mandatory data
+    await getFromCacheOrFetch({
+      query: allHomePageQuery,
+      reactQueryClient: queryClient,
+    });
+
+    // Optionnal data
+    getFromCacheOrFetch({
```

Comment:
I left a comment about that here: https://github.com/regen-network/regen-web/pull/1611#discussion_r1031618403
You got it right, basically requests with `await` block the page to render while the ones without it can be done async.

Reply:
Thanks, sorry, didn't connect the two. Makes sense.